### ISSUE TITLE
vue/v-slot-style and dynamic slot names

### ISSUE THREAD

**Issue by Tofandel** [vuejs/eslint-plugin-vue#2032]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.27
- **eslint-plugin-vue version:** 9.7

**Please show your full configuration:**

```json5
module.exports = {
  root: true,
  env: {
    browser: true,
    node: true,
    'vue/setup-compiler-macros': true,
  },
  extends: [
    'plugin:vue/vue3-recommended',
  ],
}
```

**What did you do?**	

```vue
    <template
      v-for="slot in Object.keys($slots)"
      v-slot:[slot]="scope"
    >
      <slot
        :name="slot"
        v-bind="scope"
      />
```

**What did you expect to happen?**
No eslint error or autofix to happen

**What actually happened?**

When trying to pass down slots to a subcomponent, the `vue/v-slot-style` complains that I'm not using the shortand syntax and even automatically fixes it to 
```vue
    <template
      v-for="slot in Object.keys($slots)"
      #[slot]="scope"
    >
      <slot
        :name="slot"
        v-bind="scope"
      />
```

Which doesn't compile, dynamic slot attribution only works with the `v-slot:[]` syntax

**Repository to reproduce this issue**
